### ISSUE TITLE
ETL ontology-inputs - EFO data model refactoring proposal

### ISSUE THREAD

**Issue by mbdebian** [opentargets/issues#3061]
# Discussion issue
Input files for _disease_ ETL step are located at ```ontology-inputs``` folder.

When having a look at the file _ontology-efo.jsonl_, collected by PIS from [here](https://github.com/EBISPOT/efo/releases/download/v3.57.0/efo_otar_slim.owl), according to PIS configuration [file](https://github.com/opentargets/platform-input-support/blob/150bde8f6ed472d1e3c01a8eb22ee2c6b43e92d2/config.yaml#L15), I found the following data model:
```
root
 |-- code: string (nullable = true)
 |-- dbXRefs: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- definition: string (nullable = true)
 |-- definition_alternatives: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- id: string (nullable = true)
 |-- isTherapeuticArea: boolean (nullable = true)
 |-- label: string (nullable = true)
 |-- locationIds: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- obsoleteTerms: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- parents: array (nullable = true)
 |    |-- element: string (containsNull = true)
 |-- synonyms: struct (nullable = true)
 |    |-- hasBroadSynonym: array (nullable = true)
 |    |    |-- element: string (containsNull = true)
 |    |-- hasExactSynonym: array (nullable = true)
 |    |    |-- element: string (containsNull = true)
 |    |-- hasNarrowSynonym: array (nullable = true)
 |    |    |-- element: string (containsNull = true)
 |    |-- hasRelatedSynonym: array (nullable = true)
 |    |    |-- element: string (containsNull = true)
```

Where the _synonyms_ attribute is a structure of 4 arrays: _hasBroadSynonym_, _ hasExactSynonym_, _ hasNarrowSynonym_ and _hasRelatedSynonym_

These 4 attributes are of _array_ type, not _booleans_, so I was wondering how feasible would be to promote a data model refactoring where:
- _hasBroadSynonym_ is renamed to _broad_
- _hasExactSynonym_ is renamed to _exact_
- _hasNarrowSynonym_ is renamed to _narrow_
- _hasRelatedSynonym_ is renamed to _related_
- And all _array_ types are '[]' when empty, instead of the current mix of _null_ and '[]'

**Comment by mbdebian** [comment]
@prashantuniyal02 , do we have any existing _label_ for those issues that are about proposals or meant to start a conversation?

**Comment by prashantuniyal02** [comment]
No, nothing specific. Because most issues are discussed internally first and then issues are created for specific tasks. 
I think if there are enough issues in this category, we can create a label for proposals.

**Comment by javfg** [comment]
Closing this as the rewrite of PIS and our intentions of keeping intact input files make it obsolete. We can focus on #3069.
